### ISSUE TITLE
Merge redundant linux install scripts

### ISSUE THREAD

**Issue by jerogee** [jamulussoftware/jamulus#943]
There are separate install scripts for different Linux distro's. 

```
distributions/installscripts/
├── install4debian.sh
├── install4fedora.sh
└── install4ubuntu.sh
```
I suggest to:
(1) auto-detect the distro (remove hardcoding)
(2) merge them, as they are 99% identical and mostly redundant, each containing already statements for each distro:

```
        if [ "$DISTRO"  = "Ubuntu" ]
        then
...
        elif [ "$DISTRO" = "Debian" ]
        then
etc
```

(3) if none of the 3 are detected, offer the user to install for one of the 3 anyway (with warning) or cancel the installation (default)

Happy to work on this one if there is agreement on having those merged.

**Comment by ann0see** [comment]
Yes. @niebert wrote them, @trebmuh (as far as I know) made some improvements to it in the website repo and I think we could make them a bit more visible.

Linking: https://github.com/jamulussoftware/jamuluswebsite/pull/171 and https://github.com/jamulussoftware/jamuluswebsite/blob/changes/_posts/2020-09-20-Linux-Install-Script.md

**Comment by ann0see** [comment]
I think you can work on them (at least I don't see anybody else working on them). Have a look at the script I linked in the repo which might already have a few changes you could implement.

 We might include the final scripts somewhere in the documentation repo although at least installation on debian should be easier now.

**Comment by niebert** [comment]
Sure, autodetection is a good option to maintain just one script. The only thing why I separated the scripts is to cover distro dependent packages and release dependent install calls that might differ from e.g. Ubuntu Release to the Ubuntu release Y. As long as the installer script is small and installation is almost the same then it makes sense to me to create just one script. KISS (Keep It Small and Simple) can be covered is the install scripts are create just by a build script. A JSON defines the specification for the distro and a `build_linux_install.js` generates the distro specific installation for Jamulus server and/or client. Anyway feel free to adapt the scripts according to your needs. best regards, Bert

**Comment by jerogee** [comment]
@niebert, thanks.

**Comment by ann0see** [comment]
@jerogee Any updates here? I think in the next time, the repo will be restructured (especially due to the new auto build process).

**Comment by jerogee** [comment]
@ann0see, I can send in a PR in a few days. Or will the new auto build process make these scripts redundant? (if so we should close this as "won't do")

**Comment by ann0see** [comment]
The new release process will make Debian/Ubuntu redundant.

**Comment by pljones** [comment]
It would be for the best if the automated process did everything, of course, that way we know we've a set of releasable packages for all "known" distributions at the same time.

The other benefit for getting into the automation is it's then there for good (until such time as it gets deliberately removed, of course).

**Comment by ann0see** [comment]
Do we have any agreement here?

I‘d suggest to move them (including the Raspberry Pi script) somewhere else (documentation repo/assets repo/new repo), give @niebert write access to it and link them in the documentation.

**Comment by jerogee** [comment]
@ann0see, I agree. I would vote for moving to documentation (repo) and close this issue (feel free to do so!). Otherwise we end up maintaining code for building/compiling twice: for the autobuild, and these scripts. 

Ideally, have autobuild cover what is officialy suppported and tested (compiled correctly), and have documentation describe a how-to for other (self-compile) scenarios.
